src/libsyntax/ext/tt/macro_rules.rs

```diff
@@ -285,7 +289,7 @@ fn generic_extension<'cx>(
 // Holy self-referential!
 
 /// Converts a `macro_rules!` invocation into a syntax extension.
-pub fn compile(
+pub fn compile_macro_rules(
```

Comment:
After introduction of `macro` this is more about declarative macros in general than about `macro_rules` specifically.
The comment above can be updated as well "`macro_rules!` invocation" -> "macro item".

Reply:
I think we had a discussion somewhere along the way, including the doc people, and the conclusion was to generally prefer "declarative macro" to "macro-by-example".
That was a long time ago though, when modern procedural macros were only introduced.

Reply:
I'm ok with keeping the "mbe" naming from the later commits, mostly because it's short and `mbe::TokenTree` etc. looks good, but could you add a file-level comment to `mbe.rs` explaining the abbreviation and telling what the module does?